packages/kit/src/module/install.ts

```diff
@@ -27,8 +37,8 @@ export async function installModule (moduleToInstall: string | NuxtModule, inlin
   if (typeof moduleToInstall === 'string') {
     nuxt.options.build.transpile.push(normalizeModuleTranspilePath(moduleToInstall))
     const directory = getDirectory(moduleToInstall)
-    if (directory !== moduleToInstall) {
-      nuxt.options.modulesDir.push(getDirectory(moduleToInstall))
+    if (directory !== moduleToInstall && !localLayerModuleDirs.has(directory)) {
```

Comment:
we might even be able to remove this test entirely and only add directories within `node_modules` 🤔